lib/floe/workflow/runner.rb

```diff
@@ -5,29 +5,55 @@ class Workflow
     class Runner
       include Logging
 
-      TYPES         = %w[docker podman kubernetes].freeze
       OUTPUT_MARKER = "__FLOE_OUTPUT__\n"
 
       def initialize(_options = {})
       end
 
+      @runners = {}
       class << self
-        attr_writer :docker_runner
+        # backwards compatibility -- use Floe.register_runner instead
+        def docker_runner=(value)
+          register_runner(value)
+        end
+
+        #
+        # example
+        #   register_runner "kubernetes", {}
+        #   register_runner nil # use default
+        #   register_runner Floe::Workflow::Runner::Kubernetes.new
+        #
+        #   # specify with a schema
+        #   register_runner "docker", nil # use default runner (docker)
+        #   register_runner "docker", "kubernetes", {}
+        def register_runner(schema = "docker", name_or_instance = {}, options = {})
```

Comment:
Register reads funny to me.  When i see a register method, I think of it like adding something to a registry.  So, in plugin-style libraries, I think of the register method being called for each of the various plugins to put each of them into the collective registry.

Instead this method is just picking a runner based on a symbol/string, which, to me, is more like what I'd expect from `docker_runner=` itself.  Perhaps we can just use that method directly?  Since we are < 1.0 release, I think it's ok to have a breaking change like that.

Reply:
Or perhaps `select_docker_runner` or `set_docker_runner`?

Reply:
I would prefer if floe knew about the classes associated with the runners rather than workflow or `exe/floe`.

But if we think the caller should know about these things, then we can go in that direction. I'll see if I still have the branch around that had that.

Reply:
I renamed this method. think it is what you want.